tests/test_sasa.py

```diff
@@ -149,3 +148,33 @@ def test_sasa_5():
     np.testing.assert_approx_equal(calc_area_sulfur, true_area_sulfur)
     # Finally, ensure that we are not changing _ATOMIC_RADII
     assert prev_area_chlorine != calc_area_chlorine
+
+def test_sasa_6(get_fn):
+    #Test the atom selection
+    t = md.load(get_fn('frame0.h5'))
+    atoms_resid1 = t.top.select("resid 1").astype(int)
+    atoms_resid0_2 = t.top.select("resid 0 2").astype(int)
+
+    # Mode="atom"
+    sasa_all_atoms = md.geometry.shrake_rupley(t)
+    sasa_all_atoms_w_selection = md.geometry.shrake_rupley(t, atom_indices=atoms_resid1)
+
+    # The computed SASA values are the same as in "normal" mode
+    np.testing.assert_array_almost_equal(sasa_all_atoms[:,atoms_resid1],
+                                         sasa_all_atoms_w_selection[:,atoms_resid1])
+    # The uncomputed SASA values are all np.nan
+    np.testing.assert_equal(np.unique(sasa_all_atoms_w_selection[:,atoms_resid0_2]),np.nan)
+
+    # Mode="residues"
+    sasa_all_residues = md.geometry.shrake_rupley(t, mode="residue")
+    sasa_all_residues_w_selection = md.geometry.shrake_rupley(t, atom_indices=atoms_resid1, mode="residue")
```

Comment:
Could we also make sure that the SASA values match the expected result when the other residue selection is made? It's nice that you picked a residue other than the first/zeroth for this test, and checking it for the multiple and non-sequential residue selection would be nice as well

Reply:
I've updated the tests accordingly. I have also added a test for selecting a subset of atoms (not all) of a residue, checking that e.g. if you select only one atom, mode="atom" or mode="residue" yield the same value (not that I was expecting otherwise but just to try an edge-case)